pkg/ebpf/c/tracee.bpf.c

```diff
@@ -2480,10 +2480,38 @@ static __always_inline int events_perf_submit(program_data_t *p, u32 id, long re
 
 // INTERNAL: STRINGS -------------------------------------------------------------------------------
 
+// Workaround: Newer LLVM versions might fail to optimize has_string()
+// loop unrolling with the following error:
+//
+//     warning: loop not unrolled: the optimizer was unable to perform
+//     the requested transformation; the transformation might be
+//     disabled or specified as part of an unsupported transformation
+//     ordering
+//
+
+#if defined(__clang__) && __clang_major__ > 13
+
+    #define has_prefix(p, s, n)                                                                    \
+        ({                                                                                         \
+            int rc = 0;                                                                            \
+            char *pre = p, *str = s;                                                               \
+            _Pragma("unroll") for (int z = 0; z < n; pre++, str++, z++)                            \
+            {                                                                                      \
+                if (!*pre) {                                                                       \
+                    rc = 1;                                                                        \
```

Comment:
since you don't return here, won't this cause the loop to continue?
If so, then this won't make what it supposed to do (eventualy the next statement may be true and rc=0 will be returned)

Reply:
I missed the break statement. Ended up sending the "test" version (break caused issues for clang < 14). Sending it with breaks now.